cpp/src/streaming/core/channel.cpp

```diff
@@ -23,6 +23,11 @@ coro::task<Message> Channel::receive() {
     }
 }
 
+coro::task<SpillableMessages::MessageId> Channel::receive_message_id() {
+    auto msg_id = co_await rb_.consume();
+    co_return msg_id.has_value() ? *msg_id : SpillableMessages::InvalidMessageId;
+}
```

Comment:
Like Mads, I think we don't want to expose this detail. But even if we did, carving out a special value for "no you didn't get a message" is not the right way. Use a `std::optional`.